sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowEvalPythonExec.scala

```diff
@@ -44,14 +44,17 @@ case class ArrowEvalPythonExec(udfs: Seq[PythonUDF], output: Seq[Attribute], chi
     val schemaOut = StructType.fromAttributes(output.drop(child.output.length).zipWithIndex
       .map { case (attr, i) => attr.withName(s"_$i") })
 
+    val batchedIter: Iterator[Iterator[InternalRow]] =
+      iter.grouped(conf.arrowMaxRecordsPerBatch).map(_.iterator)
+
```

Comment:
Is it necessary to group the iterator like this?

Reply:
The grouped iterator looks unnecessary. Actually you still write out the rows individually.

Reply:
I guess this is for making `ArrowPythonRunner` reusable between current pandas udf and `apply()` by taking `Iterator[Iterator[InternalRow]]` instead of `Iterator[InternalRow]` as its input. The rows in grouped iterator will be one `RecordBatch`.
I'm not sure whether it's good or not, though.

Reply:
I actually find this code doesn't work now. I will fix it.

@ueshin is right, this is to reuse `ArrowEvalPython` for both the current pandas udf and `apply()`. I basically want to lift the batching logic out of `ArrowEvalPython` so the called and decide how they want rows to be batched into `RecordBatch`. 

In the current pandas udf case, it batches it by `conf.arrowMaxRecordsPerBatch` and in `apply` it batches by one group per batch.